scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/attestation/BigDLRemoteAttestationService.scala

```diff
@@ -33,84 +34,227 @@ import akka.util.Timeout
 import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
 import spray.json.DefaultJsonProtocol._
 
-import scala.io.StdIn
+import org.json4s.jackson.Serialization
+import org.json4s._
 
-import scala.concurrent.Future
+import scala.util.parsing.json._
+
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration._
 
 import org.apache.logging.log4j.LogManager
 import scopt.OptionParser
 
 import com.intel.analytics.bigdl.ppml.attestation.verifier.SGXDCAPQuoteVerifierImpl
+import com.intel.analytics.bigdl.ppml.attestation.utils.{AttestationUtil, FileEncryptUtil}
 
 object BigDLRemoteAttestationService {
-
   implicit val system = ActorSystem("BigDLRemoteAttestationService")
   implicit val materializer = ActorMaterializer()
   implicit val executionContext = system.dispatcher
   implicit val timeout: Timeout = Timeout(100, TimeUnit.SECONDS)
 
-  final case class Quote(quote: String)
-  implicit val quoteFormat = jsonFormat1(Quote)
+  val quoteVerifier = new SGXDCAPQuoteVerifierImpl()
 
-  final case class Result(result: Int)
-  implicit val resultFormat = jsonFormat1(Result)
+  var userMap : Map[String, Any] = Map.empty
+  var policyMap : Map[String, Any] = Map.empty
 
-  val quoteVerifier = new SGXDCAPQuoteVerifierImpl()
+  def checkAppIDAndApiKey(msg: Map[String, Any]): Boolean = {
+    if (!msg.contains("app_id") || !msg.contains("api_key")) {
+      false
+    } else {
+      val appID = msg.get("app_id").mkString
+      val apiKey = msg.get("api_key").mkString
+      val userMapRes = userMap.get(appID).mkString
+      if ((userMapRes != "") && apiKey == userMapRes) {
+        true
+      } else {
+        false
+      }
+    }
+  }
 
-  def main(args: Array[String]): Unit = {
+  def enroll(): Route = {
+    val appID = UUID.randomUUID.toString
+    val apiKey = AttestationUtil.generatePassword(32)
+    userMap += (appID -> apiKey)
+    val res = "{\"app_id\":\"" + appID + ",\"api_key\":\"" + apiKey + "\"}"
+    complete(res)
+  }
+
+  def registerPolicy(msg: Map[String, Any]): Route = {
+    if (!checkAppIDAndApiKey(msg)) {
+      complete(400, "Invalid app_id and api_key.")
+    } else {
+      if (!msg.contains("TDX") || msg.get("TDX").mkString == "false") {
+        if (!msg.contains("mr_enclave") || !msg.contains("mr_signer")) {
+          complete(400, "Required parameters are not provided.")
+        }
+        val appID = msg.get("app_id").mkString
+        val mrEnclave = msg.get("mr_enclave").mkString
```

Comment:
Maybe we just need one of mrEnclave and mrSigner in most cases. If customers provide both, that's also fine.